### PR TITLE
Adds plugin support for CoffeeScript 2.x.

### PR DESCRIPTION
**Pull request by noahlange**
## Additions
- modifies Webpack config to add appropriate loaders
- adds '.coffee', '.cjsx' to resolvable extensions
- transpiles source before Babylon parsing

---

All right, here's a plugin to add CoffeeScript support to Gatsby.

A few caveats: first, CoffeeScript + React is currently a troubled combination. This plugin relies upon deprecated modules that may someday prove to be dysfunctional or otherwise deficient.

Second, the installed version of CoffeeScript is *next*. This is not optional - named exports are required for page queries to work properly.

```coffeescript
# good
export pageQuery = """
  # blah blah blah...
"""

# bad
module.exports.pageQuery = """
  # blah blah blah
"""
```
And finally, you will need to manually edit your `coffee-loader` installation and install `coffeescript` separately in your project directory to ensure that CoffeeScript@next is being loaded. The very first line of source in the former's `index.js` should be the following: note the lack of dash.

```js
var coffee = require("coffeescript");
```

When `coffee-loader` starts to support CoffeeScript 2.x this hack will no longer be necessary.

That's it. Fairly simple module. Add `gatsby-plugin-coffeescript` to your `gatsby-config.js` file and go to town.

### CODE REVIEW

File: packages/gatsby-plugin-coffeescript/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Notes
+First, note that CoffeeScript + React is a troubled combination. This plugin
+relies upon deprecated modules that may someday prove to be dysfunctional or
+otherwise deficient.
+
+Furthermore, note that the installed version of CoffeeScript is @next. This is 
+not optional - *named exports are required for page queries to work properly.*
+
+You will need to manually edit your `coffee-loader` installation and install
```
Comment (KyleAMathews) [inline packages/gatsby-plugin-coffeescript/README.md]: I don't get this part. Shouldn't this plugin handle this?

Comment (noahlange) [inline packages/gatsby-plugin-coffeescript/README.md]: It's an issue with the `coffee-loader` package - it's not (yet) set up to use 2.x (which we need for imports/exports). There's a recently-submitted PR that adds support, so this line of documentation will only be relevant for a couple days.

The repo with the PR could be installed to the project directory as `jalavosus:cs2`, and it should all work. ...that's probably be a better solution, albeit one I didn't think of until now.

https://github.com/webpack-contrib/coffee-loader/issues/33
https://github.com/webpack-contrib/coffee-loader/pull/30

Comment (KyleAMathews) [inline packages/gatsby-plugin-coffeescript/README.md]: Huh, ok, well let's just add this as-is for now and update as coffee-loader is updated.

**Comment by KyleAMathews** [comment]
Published this to NPM in 1.0.0-alpha12-alpha.1b1c742b 🎉

**Pull request merged**